### PR TITLE
Turret

### PR DESCRIPTION
**Pull request by OliverW10**
turret component that handles rotating the turret and wrapping its rotation

### CODE REVIEW

File: components/turret.py [hunk 1]
```diff
@@ -1,3 +1,99 @@
...
+        self.motor.configMotionAcceleration(self.CRUISE_ACCELERATION, 10)
+        # self.motor.configAllowableClosedloopError(0, self.ACCEPTABLE_ERROR_COUNTS, 10)
+
+        self.motor.configSelectedFeedbackSensor(
+            ctre.FeedbackDevice.CTRE_MagEncoder_Relative, 0, 10
+        )
+
+        self.motor.setSelectedSensorPosition(
+            0
+        )  # replace 0 with absolute encoder position
```
Comment (auscompgeek) [inline components/turret.py:61]: nit: put the comment before the line

File: components/turret.py [hunk 2]
```diff
@@ -1,3 +1,99 @@
...
+    def slew_local(self, angle: float) -> None:
+        """Slew to a robot relative angle"""
+        self.target = angle
+
+    @magicbot.feedback
+    def get_angle(self):
+        return self.motor.getSelectedSensorPosition() / self.COUNTS_PER_TURRET_RADIAN
+
+    def get_angle_at(self, t: float) -> float:
+        # loops_ago = int((wpilib.Timer.getFPGATimestamp() - t) / self.control_loop_wait_time)
```
Comment (auscompgeek) [inline components/turret.py:91]: Any reason this code is commented out?

**Review by auscompgeek — APPROVED** [review]
lgtm

File: components/turret.py (hunk 2 above)
Comment (OliverW10) [reply components/turret.py:91]: it was casuing errors, i will fix it in target estimator as that is the only thing that uses it

**Pull request merged**